### PR TITLE
Suggest fields/locals when specifying the loop variable in VB for loops

### PR DESCRIPTION
**Pull request by rchande**
Fixes internal bug 153633

### CODE REVIEW

**Comment by rchande** [comment]
Tagging @dotnet/roslyn-ide for review

**Comment by CyrusNajmabadi** [comment]
I'm confused.  Were these variables previously not showing up?

**Comment by rchande** [comment]
Correct.

**Comment by CyrusNajmabadi** [comment]
Wouldn't that be a bug at the compiler layer hten?  Why is LookupSymbols not returning these values?

**Comment by rchande** [comment]
No, the bug was we never called LookupSymbols after a ForKeyword.

**Comment by balajikris** [comment]
:+1:

File: src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb [hunk 1]
```diff
@@ -68,11 +68,29 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Recommendations
...
+                Dim field = DirectCast(symbol, IFieldSymbol)
+                Return Not field.IsReadOnly AndAlso Not field.IsConst
+            End If
+
+            If symbol.Kind() = SymbolKind.Local Then
+                Dim local = DirectCast(symbol, ILocalSymbol)
+                Return Not local.IsConst
+            End If
+
+            Return False
```
Comment (jasonmalinowski) [inline src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb]: We don't care about insanity where people are doing For SomeClass.SomeField = 1 To 10? I assume we aren't, just wanted to call that out.

**Comment by jasonmalinowski** [comment]
Should there be a test that we're in suggestion mode at this point? Otherwise :+1:

**Comment by rchande** [comment]
Yep, there are tests for the builder:http://source.roslyn.io/#Roslyn.Services.Editor.VisualBasic.UnitTests/Completion/CompletionProviders/SuggestionModeCompletionProviderTests.vb,201

File: src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb (hunk 1 above)
Comment (rchande) [inline src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb]: The old LS did this pretty nicely with common/all. They put writable fields/locals in the common tab and did the full list in the all tab. We can't currently do that, but I'm reluctant to just do the full list. Thoughts @DustinCampbell @Pilchie ?

Comment (DustinCampbell) [inline src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb]: Yeah, I care less about the thing @jasonmalinowski said :smile:

**Pull request merged**